src/energyplus/ForwardTranslator/ForwardTranslateAirConditionerVariableRefrigerantFlowFluidTemperatureControl.cpp

```diff
@@ -0,0 +1,362 @@
+*  INCLUDING, BUT NOT LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+*  DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER(S), ANY CONTRIBUTORS, THE UNITED STATES GOVERNMENT, OR THE UNITED
+*  STATES DEPARTMENT OF ENERGY, NOR ANY OF THEIR EMPLOYEES, BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL,
+*  EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF
+*  USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT,
+*  STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF
+*  ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+***********************************************************************************************************************/
+
+#include "../ForwardTranslator.hpp"
+#include "../../model/Model.hpp"
+
+#include "../../model/AirConditionerVariableRefrigerantFlowFluidTemperatureControl.hpp"
+#include "../../model/AirConditionerVariableRefrigerantFlowFluidTemperatureControl_Impl.hpp"
+#include "../../model/LoadingIndex.hpp"
+#include "../../model/LoadingIndex_Impl.hpp"
+
+#include "../../model/Schedule.hpp"
+#include "../../model/Schedule_Impl.hpp"
+#include "../../model/Node.hpp"
+#include "../../model/Node_Impl.hpp"
+#include "../../model/PlantLoop.hpp"
+#include "../../model/PlantLoop_Impl.hpp"
+#include "../../model/ThermalZone.hpp"
+#include "../../model/ThermalZone_Impl.hpp"
+#include "../../model/ZoneHVACTerminalUnitVariableRefrigerantFlow.hpp"
+#include "../../model/ZoneHVACTerminalUnitVariableRefrigerantFlow_Impl.hpp"
+#include "../../model/Curve.hpp"
+#include "../../model/Curve_Impl.hpp"
+#include "../../utilities/core/Logger.hpp"
+#include "../../utilities/core/Assert.hpp"
+#include <utilities/idd/AirConditioner_VariableRefrigerantFlow_FluidTemperatureControl_FieldEnums.hxx>
+#include <utilities/idd/ZoneTerminalUnitList_FieldEnums.hxx>
+#include "../../utilities/idd/IddEnums.hpp"
+#include <utilities/idd/IddEnums.hxx>
+#include "../../utilities/idf/IdfExtensibleGroup.hpp"
+#include <utilities/idd/FluidProperties_Name_FieldEnums.hxx>
+
+using namespace openstudio::model;
+
+namespace openstudio {
+
+namespace energyplus {
+
+  boost::optional<IdfObject> ForwardTranslator::translateAirConditionerVariableRefrigerantFlowFluidTemperatureControl(
+    model::AirConditionerVariableRefrigerantFlowFluidTemperatureControl& modelObject) {
+    boost::optional<std::string> s;
+    boost::optional<double> value;
+
+    IdfObject idfObject(IddObjectType::AirConditioner_VariableRefrigerantFlow_FluidTemperatureControl);
+
+    m_idfObjects.push_back(idfObject);
+
+    // Heat Pump Name
+    s = modelObject.name();
+    if (s) {
+      idfObject.setString(AirConditioner_VariableRefrigerantFlow_FluidTemperatureControlFields::HeatPumpName, *s);
+    }
```

Comment:
createNameAndRegister

Reply:
Incidentally, should we translate this AirCondVRF if it has zero terminals?

Reply:
Will createNameAndRegister have a problem with `HeatPumpName` instead of `Name`?

Reply:
A good question indeed. I had to look it up, and it'll work, because the field has a `\reference` and no `\object-list`.

Bottom line is there:

https://github.com/NREL/OpenStudio/blob/87056a8a60fc2f82fb19046141968e9c9b93567b/src/utilities/idd/IddField.cpp#L187-L190

Reply:
Should we also update the existing VRF FT to return when it has zero terminals?